### DISCUSSION TITLE
Is the result of useQuery typed?

### DISCUSSION THREAD

**Discussion by dohomi** [gqty-dev/gqty#1165] [Q&A] [answered]
Hello,

I took a quick peek at this library and have a question. After I followed the installation and got the cli installed and ran the codegen script (my introspection is a Hasura backend). I used a basic query:
```ts
const query = useQuery()

query.//=> is here supposed to be code completion based on my schema?
```
In my IDE I had the impression that I would receive code completion for the const query based on my schema. Is that supposed to be the case?

**Comment by PabloSzx** [comment]
yes, that should've worked, you should be able to inspect the types behind useQuery too

**Comment by dohomi** [comment, marked as answer]
I receive auto completion but the query is only 
```
Proxy {$state: {…}, __typename: Symbol(gqty-proxy), authProvider: Symbol(gqty-proxy), authProviderRequest: Symbol(gqty-proxy), authProviderRequests: Symbol(gqty-proxy), …}
```
inside of the queryFetcher of index.ts I add a console log but that is never fetched. Also the autogenerated file index.ts seems to require `@gqty/react` which is not mentioned in the docs yet.

Where is the place to add authentication tokens for the backend api? I thought I'd place it in queryFetcher but so far that function is not called.

**Reply by PabloSzx** [reply]
it is the query fetcher, if it's not being called is because you are not requesting the fields for gqty to go to the backend

you can't use console.log for most of gqty, you have to rely on typescript autocompletion and types to guide you since gqty works with proxies

**Reply by dohomi** [reply]
@PabloSzx I use Hasura as a backend and my queryFetcher url and authentication header is modified to point the backend.

In a react component I simply follow the basic example and call 
```

 const query = useQuery({
        suspense: false,
        staleWhileRevalidate: true,
        onError(e) {
            console.log("ERROR", e)
        }
    })

    if(query.$state.isLoading){
        return <Spinner/>
    }
     return (
        <XStack space={'$4'}>
            {query.club?.map(c => { // this breaks because club is not an array and "c" is any instead of the club fields
```
What could it be that I am missing?

**Reply by PabloSzx** [reply]
it's hard to know without a reproduction repo tbh 😕

**Reply by dohomi** [reply]
is there a quick start repo to be able to clone?

**Reply by dohomi** [reply]
@PabloSzx I found out that if I enable a table permission inside of Hasura to be visible to all, then the query works. 

But as my Hasura backend requires auth it seems not to be sufficient to set authentication headers, it seems the query runs before the authentication is resolved.

Is there any best practice how to use gqty with Hasura?
<img width="527" alt="Screenshot 2022-10-13 at 13 12 11" src="https://user-images.githubusercontent.com/489221/195507434-939878fc-66ae-4f9f-afd2-42ff9bd2e7ff.png">

**Reply by dohomi** [reply]
As soon in Hasura the permission is set to public without any authentication it works as expected. But thats not something I can do in my schema. 
<img width="1222" alt="Screenshot 2022-10-13 at 13 29 03" src="https://user-images.githubusercontent.com/489221/195509573-116298f2-9108-486b-8653-50844d2b9642.png">

**Reply by dohomi** [reply]
> it's hard to know without a reproduction repo tbh 😕

everything works as expected!.. damn had a typo in my auth header - thats why nothing worked.
